shared/api/cluster.go

```diff
@@ -0,0 +1,58 @@
+package api
+
+// Cluster represents high-level information about a LXD cluster.
+type Cluster struct {
+	StoragePools []StoragePool `json:"storage_pools" yaml:"storage_pools"`
+	Networks     []Network     `json:"networks" yaml:"networks"`
+}
```

Comment:
This should instead be more like:

```go
type ClusterMemberConfig struct {
    Name string
    Description string
}

type Cluster struct {
    Enabled bool
    Status string
    MemberConfig []ClusterLocalConfig
    CurrentServer string
}

// Very rough, will update once we sort out the right workflow there
type ClusterPut struct {
   Enabled bool
   MemberConfig map[string]string
   Cert string
   Key string
   RaftNodes ClusterRaftNodes
}
```

Reply:
As said above, ClusterMemberConfig might not be the best format for how this info is intended to be consumed.